core-web/libs/sdk/experiments/src/lib/persistence/index-db-database-handler.ts

```diff
@@ -0,0 +1,144 @@
+interface DbConfig {
+    db_name: string;
+    db_store: string;
+    db_key_path: string;
+}
+
+const DB_DEFAULT_VERSION = 1;
+
+/**
+ * The `DatabaseHandler` class offers specific methods to store and get data
+ * from IndexedDB.
+ *
+ * @example
+ * // To fetch data from the IndexedDB
+ * const data = await DatabaseHandler.getData();
+ *
+ * @example
+ * // To store an object of type AssignedExperiments to IndexedDB
+ * await DatabaseHandler.persistData(anAssignedExperiment);
+ *
+ * @example
+ * // To get an object of type AssignedExperiments to IndexedDB
+ * await DatabaseHandler.persistData(anAssignedExperiment);
+ *
+ */
+
+export class IndexDBDatabaseHandler {
+    constructor(private config: DbConfig) {
+        if (!config) {
+            throw new Error('Config is required');
+        }
+
+        const { db_name, db_store, db_key_path } = config;
+
+        if (!db_name) {
+            throw new Error("'db_name' is required in config");
+        }
+
+        if (!db_store) {
+            throw new Error("'db_store' is required in config");
+        }
+
+        if (!db_key_path) {
+            throw new Error("'db_key_path' is required in config");
+        }
```

Comment:
```javascript
const REQUIRED_KEYS = ["db_name", "db_store", "db_key_path"];

const configKeys = Object.keys(config);

REQUIRED_KEYS.forEach((key) => {
  if (!configKeys.includes(key)) {
    throw new Error(`"${key}" is required in config`);
  }
});

```

Reply:
More Cognitive Complexity for me. I want to see there what happened quickly and easily.